packages/salesforcedx-vscode-apex/src/views/testOutlineProvider.ts

```diff
@@ -111,6 +110,22 @@ export class ApexTestOutlineProvider
     this.onDidChangeTestData.fire();
   }
 
+  public async onResultFileCreate(
+    apexTestPath: string,
+    testResultFile: string
+  ) {
+    await this.refresh();
```

Comment:
Shouldn't we only refresh the sidebar if the new file matches the data in `test-run-id.txt` ?

Reply:
Yeah, I tested it and makes more sense to refresh only if the new file matches the correct data. The only difference is that now, the refresh happens immediately before the new results are displayed instead of immediately after the user starts a new test run- so it may not be _as_ apparent from the user's perspective that a refresh is occurring.